### DISCUSSION TITLE
Add glfx filters to fabricjs

### DISCUSSION THREAD

**Discussion by privyreza** [fabricjs/fabric.js#7058] [Q&A]
Hello there, Thanks for the great library!

I am struggling to create an `ink` filter based on `https://github.com/evanw/glfx.js`. Is there a way I can pass the canvas to fx and apply ink filter like soo?

```
import fx from '@/lib/glfx';

/**
 * Blur object
 * @class HueRotation
 * @extends {fabric.Image.filters.Convolute}
 * @ignore
 */
const Ink = fabric.util.createClass(
  fabric.Image.filters.BaseFilter,
  /** @lends Convolute.prototype */ {
    /**
     * Filter type
     * @param {String} type
     * @default
     */
    type: 'Ink',

    initialize(options) {
      this.ink = options.ink || 0;
    },

    applyTo(canvasEl) {
      const fxCanvas = fx.canvas(canvasEl);
      fxCanvas.ink(this.ink).update();

      ...
    },
  }
);

export default Ink;
```

Any ideas, suggestions are welcome, thanks

**Comment by asturur** [comment]
i think it would be easier to rewrite the filter shader part in a new fabricJS filter.
What does not work of the example you are showing in your post?
Where did you stop at, what errors did you encounter?
